uarray/_uarray_dispatch.cxx

```diff
@@ -227,6 +231,46 @@ PyObject * register_backend(PyObject * /*self*/, PyObject * args)
   Py_RETURN_NONE;
 }
 
+void clear_single(std::string domain, bool registered, bool global)
+{
+  if (global_domain_map.find(domain) == global_domain_map.end())
+    return;
+
+  if (registered && global)
+    {
+      global_domain_map.erase(domain);
+      return;
+    }
+  
+  if (registered)
+  {
+    global_domain_map[domain].registered.clear();
+  }
+
+  if (global)
+  {
+    global_domain_map[domain].global.backend = py_ref::ref(nullptr);
+  }
+}
+
+PyObject * clear_backends(PyObject * Py_UNUSED(self), PyObject * args)
+{
+  char* domain = nullptr;
+  int registered = true, global = false;
+  if (!PyArg_ParseTuple(args, "s|pp", 
+    &domain, &registered, &global))
```

Comment:
This won't accept domains with null characters. Just take by Object and call `domain_to_string`.

Reply:
Will domains have null characters?

Reply:
It's an arbitrary string so there's no reason it couldn't.  Better to use `domain_to_string` everywhere either way.